### PR TITLE
Include Mastodon as social media option with rel=me attribute for verificatiion

### LINKED ISSUE #18978: Verify our identity on Mastodon using `rel=me` attribute
### Discussed in https://github.com/forem/forem/discussions/18822

<div type='discussions-op-text'>

<sup>Originally posted by **erinb223** December  7, 2022</sup>
Hey team! Got a Mastodon-related quick win for ya today: profile verification.

The most effective way to verify our identity on Mastodon is to add a link with a `rel=me` attribute to both our site homepage (dev.to) and our GitHub repo (github.com/forem/forem). [Mastodon has documentation on how to do it and how it works](https://docs.joinmastodon.org/user/profile/#verification).

This is what our profile looks like currently:
A screenshot of our Mastodon profile, with all our links boxed in standard gray.

Once we have verified our ownership of our domain and our repo using the above method, they will look like these:
A screenshot of Scott Hanselman's Mastodon profile, which shows three of his links boxed in green with a checkmark.

One way to incorporate this on the dev.to homepage, while also encouraging Mastodon users to engage with us on their social media platform of choice, could be to add a Mastodon icon to our social media icon selection here:
A screenshot of our homepage with the social media icons circled in red

Verification on Mastodon is key for the following reasons:
- It proves that we are the real dev.to, not a mirror
- It makes our audience more likely to trust us on the platform
- Users can safely, confidently follow verified links off the platform and onto our sites

Thanks for your consideration! I think this could be something of a quick win and I appreciate your work on it.</div>

### PR DESCRIPTION
**Pull request by fdocr**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature

## Description

Adds Mastodon as an option for social media in `/admin/customization/config` and includes a `rel=me` attribute for verification ([read more here](https://docs.joinmastodon.org/user/profile/#verification))

## Related Tickets & Documents

- Closes #18978

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Add a Mastodon URL on `/admin/customization/config` under `Social Media` tab
3. Check the main feed to ensure it's there and the link works properly
4. Add all possible social media options
5. Check the icons don't overflow on the main feed

**Before:**

Screen Shot 2023-01-26 at 11 52 22

**After:**

Screen Shot 2023-01-26 at 11 52 36

### UI accessibility concerns?

No accessibility concerns, but a small detail to note is that when 6 social media options are enabled the icons overflow. Because of this a small margin is removed. I'm unsure if this is problematic but wanted to make note of it in case others have a suggestion to handle this in a different/better way 🙂 

## Added/updated tests?

- [x] No, and this is why: Existing settings and other specs should cover the existing functionality

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?

Enable the Mastodon Social Media option on DEV

## [optional] What gif best describes this PR or how it makes you feel?

Elephant sliding down a hill

### CODE REVIEW

File: app/views/admin/settings/forms/_rate_limit.html.erb [hunk 1]
```diff
@@ -63,12 +63,12 @@
...
               <%= admin_config_label platform %>
               <p class="crayons-field__description">
-              <%= platform.capitalize %> username
+              <%= platform == "mastodon" ? "Mastodon complete URL including \"https://\"" : "#{platform.capitalize} username" %>
               </p>
               <%= social_media_field.text_field platform,
                                                 class: "crayons-textfield",
                                                 value: Settings::General.social_media_handles[platform],
-                                                placeholder: "" %>
+                                                placeholder: platform == "mastodon" ? "https://fosstodon.org/@thepracticaldev" : "" %>
```
Comment (fdocr) [inline app/views/admin/settings/forms/_rate_limit.html.erb:71]: Mastodon is different so it requires a full URL to work

File: app/views/layouts/_social_media.html.erb [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+  <%# There isn't a single Mastodon site to build the target URL so the setting should contain the complete URL %>
+  <% target_url = social_media_type == "mastodon" ? handle : "https://#{social_media_type}.com/#{handle}" %>
   <a
-    href="<%= "https://#{social_media_type}.com/#{handle}" %>"
+    href="<%= target_url %>"
     target="_blank"
-    class="c-link c-link--icon-alone c-link--block mx-1"
-    rel="noopener">
+    class="c-link c-link--icon-alone c-link--block <%= compact_margin ? "" : "mx-1" %>"
+    rel="noopener me">
```
Comment (fdocr) [inline app/views/layouts/_social_media.html.erb:10]: `compact_margin` here essentially gets rid of the `mx-1` padding so the icons don't overflow (only in the case more than 5 social media links are enabled)

File: app/views/admin/settings/forms/_rate_limit.html.erb (hunk 1 above)
Comment (jaw6) [reply app/views/admin/settings/forms/_rate_limit.html.erb:71]: Not blocking, but just curious: I was initially quite confused why this PR was making changes in what seemed to be an unrelated area, until I dug a bit and realized what was going on -- would it be worth separating the social media settings from the `rate_limit` partial?

**Review by jaw6 — APPROVED** [review]

**Review by Ridhwana — APPROVED** [review]

File: app/views/admin/settings/forms/_rate_limit.html.erb (hunk 1 above)
Comment (fdocr) [reply app/views/admin/settings/forms/_rate_limit.html.erb]: Makes sense to me too, great suggestion @jaw6 💯

**Pull request merged**